### ISSUE TITLE
HTML cell contents are incorrectly rendered in the filter

### ISSUE THREAD

**Issue by coderazzi** [coderazzi/tablefilter-swing#69]
(Original issue 23 created by {u'display_name': u'altmany', u'account_id': None} on 2011-12-12T15:55:46+00:00)

```
What steps will reproduce the problem?
1. Create a JTable with some HTML cell contents. For example: &quot;&lt;html&gt;&lt;b&gt;bold
text&quot;
2. Notice that the cell displays the cell contents correctly (i.e., as a bold text
in this example). This is because JTables accept HTML strings, just as in other Swing
components.
3. Apply a filter with auto choices to this table

What is the expected output? What do you see instead?
The filter displays the data as &quot;\&lt;html&gt;&lt;b&gt;bold text&quot; instead
of rendering it as an HTML-formatted label. This makes the filter ugly and unusable
for any table column that contains HTML contents.

What version of the product are you using? On what operating system?
4.2.0 on JVM 1.6.0u17, WinXP

Please provide any additional information below.
Notice that if I use CustomChoices instead of the AutoChoices, then the HTML string
*does* render as expected (i.e., with all the relevant HTML formatting).

```

Original issue reported on code.google.com by `altmany` on 2011-12-12 15:55:46

**Comment by coderazzi** [comment]
(Original issue 23 last updated on 2011-12-12T15:55:46+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2011-12-17T09:12:26+00:00)

```
Hi,

Yep, I see the problem; the reason why the filter displays it as '\<html>....' is because
when it is evaluated, without the initial backslash it would interpretate it as the
minus operator.

In addition, the default filter works as a regular expression matcher, so it is totally
oblivious to html content -would work in most of the cases, anyway.

I will check how to better implement it,

Cheers,

  Lu.  
```

Original issue reported on code.google.com by `coderazzi` on 2011-12-17 09:12:26

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2012-01-16T22:44:46+00:00)

```
Fixed on release 4.3.0
```

Original issue reported on code.google.com by `coderazzi` on 2012-01-16 22:44:46
